packages/frameworks/vue/src/menu/use-menu.ts

```diff
@@ -1,43 +1,63 @@
 import * as menu from '@zag-js/menu'
 import { normalizeProps, useMachine, type PropTypes } from '@zag-js/vue'
-import { computed, reactive, type ComputedRef } from 'vue'
+import { computed, watch, type ComputedRef } from 'vue'
 import { useEnvironmentContext } from '../environment'
 import type { Optional } from '../types'
 import { useId } from '../utils'
 
-export type UseCarouselContext = Optional<menu.Context, 'id'>
+export interface UseMenuProps extends Optional<menu.Context, 'id'> {
+  modelValue?: menu.Context['value']
+}
 
-export type UseMenuReturn = {
+export interface UseMenuReturn {
   api: ComputedRef<menu.Api<PropTypes>>
-  menuMachine: ReturnType<typeof menu.machine>
+  machine: ReturnType<typeof menu.machine>
 }
 
-export const useMenu = (emit: CallableFunction, context: UseCarouselContext): UseMenuReturn => {
-  const reactiveContext = reactive(context)
-
+export const useMenu = (props: UseMenuProps, emit: CallableFunction): UseMenuReturn => {
+  const context = computed(() => {
+    const { modelValue, ...rest } = props
+    return {
+      ...rest,
+      value: modelValue,
+    }
+  })
   const getRootNode = useEnvironmentContext()
 
-  const [state, send, menuMachine] = useMachine(
+  const [state, send, machine] = useMachine(
     menu.machine({
-      ...reactiveContext,
-      id: reactiveContext.id || useId().value,
+      ...context.value,
+      id: context.value.id || useId().value,
       getRootNode,
-      onOpenChange() {
-        emit('open-change')
+      onOpenChange(details) {
```

Comment:
```suggestion
      onOpenChange: (details) => {
```
I hope it doesn't break anything. We use arrow functions for emitters.